tests/epyccel/test_epyccel_decorators.py

```diff
@@ -146,3 +146,28 @@ def get_val(x : int = None , y : int = None):
     g = epyccel(f, language=language)
 
     assert f() == g()
+
+def test_indexed_template(language):
+    @template(name='T', types=[float, complex])
+    def my_sum(v: 'T[:]'):
+        return v.sum()
+
+    pyccel_sum = epyccel(my_sum, language=language)
+
+    x = np.ones(4, dtype=float)
+    y = np.ones(4, dtype=float)
+
+    python_fl = my_sum(x)
+    pyccel_fl = pyccel_sum(y)
+
+    assert python_fl == pyccel_fl
+    assert isinstance(python_fl, type(pyccel_fl))
+
+    x = np.ones(4, dtype=complex)
+    y = np.ones(4, dtype=complex)
```

Comment:
Maybe we should have a proper complex test?
```suggestion
    x = np.ones(4, dtype=complex) * (1 + 3j)
    y = np.ones(4, dtype=complex)
```

Reply:
Fixed f0cd8cc6